backend/debug/compute_ranges.ml

```diff
@@ -251,33 +286,51 @@ module Make (S : Compute_ranges_intf.S_functor) = struct
             (key :: S.Key.all_parents key))
         case result
     in
-    let actions =
-      (* Ranges must be closed before they are opened---otherwise, when a
-         variable moves between registers at a range boundary, we might end up
-         with no open range for that variable. Note that the pipeline below
-         constructs the [actions] list in reverse order---later functions in the
-         pipeline produce actions nearer the head of the list. *)
+    (* Ranges must be closed before they are opened---otherwise, when a variable
+       moves between registers at a range boundary, we might end up with no open
+       range for that variable. Note that the pipeline below constructs the
+       [actions] list in reverse order---later functions in the pipeline produce
+       actions nearer the head of the list. *)
+    []
+    |> handle case_1b Open_subrange_one_byte_after
+    |> handle case_1c Open_one_byte_subrange
+    |> handle case_1d Open_subrange
+    |> handle case_2a Close_subrange
+    |> handle case_2b Open_subrange_one_byte_after
+    |> handle case_2b Close_subrange
+    |> handle case_2c Close_subrange_one_byte_after
+
+  let actions_at_instruction ~(insn : L.instruction)
+      ~(prev_insn : L.instruction option) ~known_available_after_prev_insn =
+    let available_before = S.available_before insn in
+    let available_across = S.available_across insn in
+    if debug
+    then
+      Format.eprintf "canonicalised available_before:@ %a\n"
+        (Misc.Stdlib.Option.print KS.print)
+        available_before;
+    if debug
+    then
+      Format.eprintf "canonicalised available_across:@ %a\n"
+        (Misc.Stdlib.Option.print KS.print)
+        available_across;
+    match available_before, available_across with
+    | None, None | None, Some _ | Some _, None ->
+      (* If availability information isn't known for the current instruction,
+         just skip to the next instruction. *)
+      (* CR mshinwell: maybe we could try to compute [available_across]? In what
```

Comment:
This comment looks a bit out of place now. I think removing it is fine, otherwise it needs to be expanded to explain in which case this might be possible.

Reply:
Removed